malcolm/modules/ADCore/parts/hdfwriterpart.py

```diff
@@ -516,38 +522,62 @@ def _check_xml_is_valid(self, child):
     @add_call_types
     def on_seek(
         self,
+        context: scanning.hooks.AContext,
         completed_steps: scanning.hooks.ACompletedSteps,
         steps_to_do: scanning.hooks.AStepsToDo,
     ) -> None:
         # This is rewinding or setting up for another batch, so the detector
         # will skip to a uniqueID that has not been produced yet
-        self.uniqueid_offset = completed_steps - self.done_when_reaches
-        self.done_when_reaches += steps_to_do
+        self.uniqueid_offset = completed_steps - self.last_expected_unique_id
+        self.last_expected_unique_id += steps_to_do
+        # Update the first array future
+        child = context.block_view(self.mri)
+        self.first_array_future = child.when_value_matches_async(
+            "arrayCounterReadback",
+            greater_than_value,
```

Comment:
We could zero the arrayCounter here if that makes the logic easier?

Reply:
I think this is a great idea, would allow me to remove some other attributes too